Dockerfile

```diff
@@ -46,6 +50,41 @@ RUN cd /tmp && tar -xzf /tmp/tar123.tar.gz
 RUN rm -f /bin/tar /bin/gtar
 RUN cd /tmp/tar-1.23 && FORCE_UNSAFE_CONFIGURE=1 ./configure --prefix=/ && make && make install && ln -sf /bin/tar /bin/gtar
 
+RUN curl -o /tmp/openssl-1.0.1r.tar.gz http://artfiles.org/openssl.org/source/openssl-1.0.1r.tar.gz
+RUN cd /tmp && tar -xzf /tmp/openssl-1.0.1r.tar.gz
+RUN cd /tmp/openssl-1.0.1r && ./Configure linux-x86_64 no-shared --openssldir=/opt/openssl -fPIC && make && make install
+
+# now build git
+# dependencies
+RUN yum -y install \
+    expat-devel \
+    gettext-devel \
+    perl-devel \
+    zlib-devel
+
+RUN curl -o /tmp/curl-7.46.0.tar.gz http://curl.askapache.com/download/curl-7.46.0.tar.gz
+RUN cd /tmp && tar -xzf /tmp/curl-7.46.0.tar.gz
+RUN cd /tmp/curl-7.46.0 && LIBS="-ldl" ./configure --enable-static --prefix=/opt/curl --with-ssl=/opt/openssl && make all && make install
+
+RUN /opt/curl/bin/curl -L -o /tmp/curl.tar.gz https://github.com/bagder/curl/releases/download/curl-7_46_0/curl-7.46.0.tar.gz
+RUN cd /tmp && tar xvf /tmp/curl.tar.gz
```

Comment:
clean these two lines up, just had them to verify curl was compiled successfully.